src/Crypto/RNCryptor/Types.hs

```diff
@@ -70,9 +73,21 @@ randomSaltIO :: Int -> IO ByteString
 randomSaltIO sz = C8.pack <$> forM [1 .. sz] (const $ randomRIO ('\NUL', '\255'))
 
 --------------------------------------------------------------------------------
+makeKey :: ByteString -> ByteString -> ByteString
+makeKey = generate (prfHMAC (undefined::SHA1)) (Parameters 10000 32) -- userKey salt
```

Comment:
re: use of undefined in comment above:

Agree this looks suspicious!  Copied from the pattern on cryptonite tutorial:

https://hackage.haskell.org/package/cryptonite-0.19/docs/Crypto-Tutorial.html

 ```
   ...
    ctx = cipherInitNoErr (cipherMakeKey (undefined :: AES256) secret)
   ...
 
```
Without this pattern, compiler complains:

/Users/tom_titchener/Documents/Dev/Haskell/rncryptor-hs/src/Crypto/RNCryptor/Types.hs:77:46:
    Not in scope: data constructor ‘SHA1’

/Users/tom_titchener/Documents/Dev/Haskell/rncryptor-hs/src/Crypto/RNCryptor/Types.hs:139:36:
    Not in scope: data constructor ‘AES256’

If there's a better pattern, I'd be happy to use it.  I agree this and other places where I put annotations seems like I'm missing something.